packages/client/src/Config.ts

```diff
@@ -23,9 +22,120 @@ export interface PrivateKeyAuthConfig {
+     * WebRTC connections.
+     *
+     * Probing private addresses can trigger false-positive incidents in
+     * some port scanning detection systems employed by web hosting
+     * providers. Disallowing private addresses may prevent direct
+     * connections from being formed between nodes using IPv4 addresses
+     * on a local network.
+     *
+     * Details: https://github.com/streamr-dev/network/wiki/WebRTC-private-addresses
+    */
+    webrtcAllowPrivateAddresses?: boolean
+
+    /**
+     * Defines WebRTC connection establishment timeout in milliseconds.
+     *
+     * When attempting to form a new connection, if not established within
+     * this timeout, the attempt is considered as failed and further
+     * waiting for it will cease.
+    */
+    webrtcNewConnectionTimeout?: number
+
+    /**
+     * Sets the low-water mark used by send buffers of WebRTC connections.
+    */
+    webrtcDatachannelBufferThresholdLow?: number
+
+    /**
+     * Sets the high-water mark used by send buffers of WebRTC connections.
+    */
+    webrtcDatachannelBufferThresholdHigh?: number
+
+    /**
+     * Contains connectivity information to the client's Network Node, used in the network layer.
+     * Can be used in cases where the client's public IP address is known before
+     * starting the network node. If not specified, the PeerDescriptor will be auto-generated.
+    */
+    peerDescriptor?: JsonPeerDescriptor
+
+    /**
+     * The port to use for the client's Network Node WebSocket server.
+     * If not specified, the server will not be started.
+     * The server is used by the network layer to accept incoming connections
+     * over the public internet to improve the network node's connectivity.
+     */
+    webSocketPort?: number
+}
+
+export interface NetworkNodeConfig {
+
+    /** The Ethereum address of the node. */
+    id?: string
+
+    /**
+     * The number of connections the client's network node should have
+     * on each stream partition.
+    */
+    streamPartitionNumOfNeighbors?: number
+
+    /**
+     * The minimum number of peers in a stream partition that the client's network node
+     * will attempt to propagate messages to
+     */
+    streamPartitionMinPropagationTargets?: number
+
+    /**
+     * The waited time for the first connection to be formed when first connecting
+     * to the network. If the connection is not formed within this time, the client's
+     * network node will throw an error.
+     */
+    firstConnectionTimeout?: number
+
+    /**
+     * Whether to accept proxy connections. Enabling this option allows
+     * this network node to act as proxy on behalf of other nodes / clients.
+    */
+    acceptProxyConnections?: boolean
+}
+
+export interface NetworkConfig {
+    controlLayer?: ControlLayerConfig
+    node?: NetworkNodeConfig
+}
+
+export enum JsonNodeType {
+    NODEJS = 'nodejs',
+    BROWSER = 'browser'
+}
```

Comment:
```suggestion
export type JsonNodeType = 'nodejs' | 'browser'
```

In client config all enums are string literals instead of TypeScript enums. (We try to avoid using enums elsewhere, too)